samples: Bluetooth: Audio sample board files cleanup

Thalley opened the pull request:
Clean up some of the board specific configuration files.
Removed all CONFIG_BT_CTLR* configs from the nRF5340
appcore conf files as that does not compile the controller.

Ensured that the ISO buffer in the controller for
nRF52840 can contain SDUs from all BAP LC3 minimum
requirements, as the controller does not yet
support segmentation of SDUs.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

fixes https://github.com/zephyrproject-rtos/zephyr/issues/45543

Thalley committed:
samples: Bluetooth: Audio sample board files cleanup

Clean up some of the board specific configuration files.
Removed all CONFIG_BT_CTLR* configs from the nRF5340
appcore conf files as that does not compile the controller.

Ensured that the ISO buffer in the controller for
nRF52840 can contain SDUs from all BAP LC3 minimum
requirements, as the controller does not yet
support segmentation of SDUs.

Also fixed a few bad configurations (broadcast sink not
enabling the sync support etc.).

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Casper-Bonde-Bose commented on samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf:
```diff
@@ -0,0 +1,15 @@
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_CENTRAL_ISO=y
+
+# For LC3 the following configs are needed
+CONFIG_FPU=y
+CONFIG_LIBLC3CODEC=y
+# The LC3 codec uses a large amount of stack. This app runs the codec in the work-queue, hence
+# inctease stack size for that thread.
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
+# LC3 lib requires floating point support in the c-lib NEWLIB is one way of getting that.
+CONFIG_NEWLIB_LIBC=y
```
I did not test LC3 can run on the nrf52 device with regards to CPU resources and RAM.

Thalley replied:
Hmm, good point. Didn't actually test that, but can give it a try before merging this :)

Thalley replied:
Hmm, actually I do not have a way of testing LC3 on the nRF52840 at this point, so will remove the LC3 from the board files.

Casper-Bonde-Bose reviewed (dismissed):
LGTM if someone else verified LC3 can run on the nRF5280

cvinayak commented on samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf:
```diff
@@ -1,2 +1,7 @@
-CONFIG_BT_CTLR_ADV_PERIODIC=y
+CONFIG_BT_CTLR_SYNC_PERIODIC=y
+CONFIG_BT_CTLR_SYNC_ISO=y
+
+# Supports the highest SDU size required by any BAP LC3 presets (155)
+CONFIG_BT_CTLR_ADV_ISO_PDU_LEN_MAX=155
+CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE=155
```
```suggestion
CONFIG_BT_CTLR_SYNC_ISO_PDU_LEN_MAX=155
```

These are only needed for synchronized receiver, not required for broadcaster.

cvinayak commented on samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf:
```diff
@@ -1,2 +1,6 @@
+CONFIG_BT_CTLR_PERIPHERAL_ISO=y
+
+# Supports the highest SDU size required by any BAP LC3 presets (155)
+CONFIG_BT_CTLR_ADV_ISO_PDU_LEN_MAX=155
```
```suggestion
```

cvinayak commented on samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf:
```diff
@@ -0,0 +1,6 @@
+CONFIG_BT_CTLR_CENTRAL_ISO=y
+
+# Supports the highest SDU size required by any BAP LC3 presets (155)
+CONFIG_BT_CTLR_ADV_ISO_PDU_LEN_MAX=155
```
```suggestion
```

asbjornsabo reviewed (dismissed):
LGTM

Thalley committed:
samples: Bluetooth: Broadcast audio scan and adv data size increase

Increases the scan and adv data sizes to better support
the extended advertising data and periodic advertising
data (BASE).

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Added commit to increase scan and adv data sizes.

Thalley commented:
Added commit to increase scan and adv data sizes.

cvinayak commented on samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf:
```diff
@@ -4,3 +4,5 @@ CONFIG_BT_CTLR_SYNC_ISO=y
 
 # Supports the highest SDU size required by any BAP LC3 presets (155)
 CONFIG_BT_CTLR_SYNC_ISO_PDU_LEN_MAX=155
+# Supports the highest advertising data that can be set in a single HCI command
+CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX=251
```
Should it be that 252 bytes be supported?

Thalley replied:
Chose 251 specifically as I set `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=251` because that is largest extended advertising value for a single HCI command (and surely is enough for this sample, and I think it makes more sense if the two values in the pair of samples are equal. 

I did consider 252 to support the maximum single periodic advertising report, but felt that the 251 would be better to cover both extended and periodic advertising

Thalley commented:
Rebased to solve merge conflicts

cvinayak commented on samples/bluetooth/broadcast_audio_source/boards/nrf52840dk_nrf52840.conf:
```diff
@@ -1 +1,5 @@
 CONFIG_BT_CTLR_ADV_ISO=y
+# Supports the highest SDU size required by any BAP LC3 presets (155)
+CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE=155
+# Supports the highest advertising data that can be set in a single HCI command
+CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=251
```
```suggestion
# Zephyr Controller tested maximum advertising data that can be set in a single HCI command
CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=191
```

Thalley replied:
Hmm, so we don't always support The maximum length of a single HCI command? 

In the BSIM tests we actually set `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=255` I see, so isn't it tested at 251 already?

cvinayak replied:
If a periodic advertising is associated and/or a channel classification is requested the common extended header format size increases, hence controller is tested only to 191 bytes as supported AD data length max.

The `bsim_test_audio` test should reduce `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX` to 191 bytes. Hope 191 is sufficient for audio usecases (otherwise controller has to support AD data fragmentation into multiple on-air PDU).

Thalley commented:
@carlescufi Can you please merge this? Will close 2 opened issues.
